src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -213,11 +214,42 @@ ucs_status_t uct_ib_mlx5_devx_create_qp(uct_ib_iface_t *iface,
 }
 
 ucs_status_t uct_ib_mlx5_devx_modify_qp(uct_ib_mlx5_qp_t *qp,
-                                        enum ibv_qp_state state)
+                                        const void *in, size_t inlen,
+                                        void *out, size_t outlen)
+{
+    int ret;
+
+    switch (qp->type) {
+    case UCT_IB_MLX5_OBJ_TYPE_VERBS:
+        ret = mlx5dv_devx_qp_modify(qp->verbs.qp, in, inlen, out, outlen);
+        if (ret) {
+            ucs_error("mlx5dv_devx_qp_modify(%x) failed, syndrome %x: %m",
+                      UCT_IB_MLX5DV_GET(modify_qp_in, in, opcode),
+                      UCT_IB_MLX5DV_GET(modify_qp_out, out, syndrome));
+            return UCS_ERR_IO_ERROR;
+        }
+        break;
+    case UCT_IB_MLX5_OBJ_TYPE_DEVX:
+        ret = mlx5dv_devx_obj_modify(qp->devx.obj, in, inlen, out, outlen);
+        if (ret) {
+            ucs_error("mlx5dv_devx_obj_modify(%x) failed, syndrome %x: %m",
+                      UCT_IB_MLX5DV_GET(modify_qp_in, in, opcode),
+                      UCT_IB_MLX5DV_GET(modify_qp_out, out, syndrome));
+            return UCS_ERR_IO_ERROR;
+        }
+        break;
+    case UCT_IB_MLX5_OBJ_TYPE_LAST:
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    return UCS_OK;
```

Comment:
coult move this "return UCS_OK" inside 'case', instead of 'break'

Reply:
but we still need return at the end of non void function...

Reply:
right, we consider changing "case UCT_IB_MLX5_OBJ_TYPE_LAST' to "default". but not critical